src/uhttp.c

```diff
@@ -320,7 +325,48 @@ static int write_text_line(HTTP_CLIENT_HANDLE_DATA* http_data, const char* text_
         result = 0;
         if (http_data->trace_on)
         {
-            LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+            char* authStart = strstr(text_line, AUTH_HEADER);
+            char* proxyAuthStart = strstr(text_line, PROXY_AUTH_HEADER);
+            char* authEol = strstr(authStart, HEADER_ENDING);
+            char* proxyAuthEol = strstr(proxyAuthStart, HEADER_ENDING);
+            if (authStart == NULL && proxyAuthStart == NULL)
+            {
+                LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+            }
+            else if (authStart != NULL && proxyAuthStart == NULL)
+            {
+                if (authEol != NULL)
+                {                                                                 
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s ************** %.*s", (int)(authStart  - text_line), text_line, (int)(strlen(text_line) - (authEol - text_line)), authEol);
+                }
+            }
+            else if (proxyAuthStart != NULL && authStart == NULL)
+            {
+                if (proxyAuthEol != NULL)
+                {
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s ************** %.*s", (int)(proxyAuthStart  - text_line), text_line, (int)(strlen(text_line) - (proxyAuthEol - text_line)), proxyAuthEol);
+                }
+            }
+            else
+            {
+                int authPos = authStart - text_line;
+                int proxyAuthPos = proxyAuthStart - text_line;
+                if(authPos < proxyAuthPos)
+                {
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s ***** %.*s ***** %.*s", (int)(authStart - text_line), text_line, (int)(proxyAuthStart - authEol), authEol,
+(int)(proxyAuthEol - text_line),
+proxyAuthEol);
+                }
+                else
+                {
+                    char* secondAuthStart = strstr(text_line, AUTH_HEADER_TWO);
```

Comment:
why this search again? you already did that at the start. just swap how the output is formatted

Reply:
Because the first "Authorization" is picked up in "Proxy-Authorization:", and it messes things up. That's why I search for it with AUTH_HEADER_TWO, which includes a '\n' at the beginning.